### ISSUE TITLE
Doesn't catch errors inside callbacks

### ISSUE THREAD

**Issue by IanVS** [michaelrhodes/tape-catch#1]
With the test:

```js
var test = require('tape-catch')

test('cause an exception', function (assert) {
  setTimeout(function () {
    assert.ok(asdf);
  }, 100);
})

test('still run this test', function (assert) {
  assert.equal(1 + 1, 2, 'this still ran')
  assert.end()
})
```

I get:

```
TAP version 13
# cause an exception
/home/ian/Local_code/sails-hook-jsonapi/tests/responses/test.js:5
    assert.ok(asdf);
              ^
ReferenceError: asdf is not defined
    at null._onTimeout (/home/ian/Local_code/sails-hook-jsonapi/tests/responses/test.js:5:15)
    at Timer.listOnTimeout (timers.js:89:15)
```

**Comment by ronkorving** [comment]
+1 on this. Catching uncaught exceptions would be really nice to have. Perhaps instead of wrapping the test function inside try/catch, register/unregister an uncaughtException handler and use that for the duration of the test?

**Comment by michaelrhodes** [comment]
It took a bit of messing around, but I believe I’ve resolved this issue as of `1.0.5`
